sources/networkmanager/site.go

```diff
@@ -0,0 +1,139 @@
+		Attributes:      attributes,
+		Tags:            tagsToMap(site.Tags),
+		LinkedItemQueries: []*sdp.LinkedItemQuery{
+			{
+				Query: &sdp.Query{
+					// +overmind:link networkmanager-global-network
+					// Search for all sites with this global network
+					Type:   "networkmanager-global-network",
+					Method: sdp.QueryMethod_GET,
+					Query:  *site.GlobalNetworkId,
+					Scope:  scope,
+				},
+				BlastPropagation: &sdp.BlastPropagation{
+					// ?? Sites can affect the global network
+					In: true,
+					// The global network will definitely affect the site
+					// instances
+					Out: true,
+				},
+			},
+		},
+	}
+
+	switch site.State {
+	case types.SiteStatePending:
+		item.Health = sdp.Health_HEALTH_PENDING.Enum()
+	case types.SiteStateAvailable:
+		item.Health = sdp.Health_HEALTH_OK.Enum()
+	case types.SiteStateUpdating:
+		item.Health = sdp.Health_HEALTH_PENDING.Enum()
+	case types.SiteStateDeleting:
+		item.Health = sdp.Health_HEALTH_PENDING.Enum()
+	}
+
+	return &item, nil
+}
+
+//go:generate docgen ../../docs-data
+// +overmind:type networkmanager-site
+// +overmind:descriptiveType Networkmanager Site
+// +overmind:get Get a Networkmanager Site
+// +overmind:list List a Networkmanager Sites
+// +overmind:group AWS
+// +overmind:terraform:queryMap aws_networkmanager_site.id
+
+func NewSiteSource(config aws.Config, accountID string, limit *sources.LimitBucket) *sources.AlwaysGetSource[*networkmanager.GetSitesInput, *networkmanager.GetSitesOutput, *networkmanager.GetSitesInput, *networkmanager.GetSitesOutput, NetworkmanagerClient, *networkmanager.Options] {
+	return &sources.AlwaysGetSource[*networkmanager.GetSitesInput, *networkmanager.GetSitesOutput, *networkmanager.GetSitesInput, *networkmanager.GetSitesOutput, NetworkmanagerClient, *networkmanager.Options]{
+		Client:    networkmanager.NewFromConfig(config),
+		AccountID: accountID,
+		ItemType:  "networkmanager-sites",
+		GetFunc:   sitesGetFunc,
+		GetInputMapper: func(scope, query string) *networkmanager.GetSitesInput {
+			// We are using a custom id of {globalNetworkId}/{siteId} e.g.
+			sections := strings.Split(query, "/")
+
+			if len(sections) != 2 {
+				return nil
+			}
+			return &networkmanager.GetSitesInput{
+				GlobalNetworkId: &sections[0],
+				SiteIds: []string{
+					sections[1],
+				},
+			}
+		},
+		ListInput: &networkmanager.GetSitesInput{},
+		ListFuncPaginatorBuilder: func(client NetworkmanagerClient, input *networkmanager.GetSitesInput) sources.Paginator[*networkmanager.GetSitesOutput, *networkmanager.Options] {
+			// GlobalNetworkId is required
+			// SiteIds must be valid slice, not nil
+			if input.GlobalNetworkId == nil {
+				input.GlobalNetworkId = aws.String("")
+				input.SiteIds = []string{}
+			}
+			return networkmanager.NewGetSitesPaginator(client, input)
+		},
+		ListFuncOutputMapper: func(output *networkmanager.GetSitesOutput, input *networkmanager.GetSitesInput) ([]*networkmanager.GetSitesInput, error) {
```

Comment:
This one looks similar to reource policy in that it could be onverted to a `DescribeOnlySource` to avoid needing to run too many API requests

Reply:
Well, resource policy is `GetListSource`, but site might be better to use `DescribeOnlySource `, as method `GetSites()` returns similar output to `Describe...()` methods, e.g. iterator and so on. Am I right here?